pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -465,8 +433,6 @@ func (p *PriorityQueue) Update(oldPod, newPod *v1.Pod) error {
 	if usPodInfo := p.unschedulableQ.get(newPod); usPodInfo != nil {
 		p.nominatedPods.update(oldPod, newPod)
 		if isPodUpdated(oldPod, newPod) {
-			// If the pod is updated reset backoff
-			p.clearPodBackoff(newPod)
```

Comment:
(not quite related the deleted lines)

In the update case, should we reset the backoff expiry? If yes, the current logic doesn't work as in `updatePod()`, it actually just copy newPod (*v1.Pod) to oldPodInfo and return. In other words, oldPodInfo.attempts are kept and hence be honored when calculating the backoff timeout.

cc @alculquicondor

Reply:
To preserve the current behavior, we would reset Attempts to zero. And we should document this in the comment of `Update`. Agreed?

Reply:
It's the tricky part. On one hand, to preserve the "Update" semantics, we should look at it as a "new" pod, hence reset its backoff timer. However, on the other hand, to achieve that, it seems the only way is to reset `podInfo.Attempts`, and that sort of breaks the semantics of "attempts" - in a metrics point of view, "attempts" records the number of a Pod's lifespan scheduling attempts :)

Reply:
uhm... metrics is something we should be careful with.
An alternative would be to reset Timestamp with the next backoff timeout. That way, it immediately goes to activeQ, but if would preserve its backoff. And I believe this is reasonable. Specially if there were more than one pod waiting for its affinity to be satisfied.

@ahg-g any thoughts?

Reply:
> An alternative would be to reset Timestamp with the next backoff timeout. That way, it immediately goes to activeQ, but if would preserve its backoff.

Are you referring to changing the semantics of `podInfo.TimeStamp`? If so, I don't think it's necessary.

Basically, the key point I want to raise is:

- With this PR, for every case (update or whatever others), an existing Pod's backoff is kept
- In the before, e.g., for the Update case, as we called `clearPodBackoff`, the updated pod was (1) put back into activeQ, and (2) its backoff timer was reset.

So we should discuss the scenarios of resetting backoff timer here, whether we should, or shouldn't, e.g., for the Update case.

Reply:
A couple of things:
1) It seems that we are preserving the semantics of the backoff start time (```podInfo.Timestamp``` is updated when ```podLastUpdateTime``` used to be updated), this is good...

2) Number of ```podInfo.Attempts``` doesn't have the same semantics as ```podAttempts``` map, the latter gets reset on update, the former doesn't.

Two options:
1)  We add ```backoffAttempts``` to podInfo to maintain the current semantics . 
2) On update, the pod is placed in the activeQ directly (line 478 below), so it is getting an immediate chance for reevaluation. So, maybe it is not so bad if we assume that the pod should not have its attempts reset to 0 when calculating the next backoff if the pod didn't get scheduled.

Reply:
> So, maybe it is not so bad if we assume that the pod should not have its attempts reset to 0 when calculating the next backoff if the pod didn't get scheduled.

It's exactly the point I want to get into a consensus. I'm fine with this option.

Reply:
I think this is reasonable.

Reply:
@notpad was there a unit test for this? If not, please add one.